clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -100,46 +100,61 @@
+            + "Kafka's group management facility. The client sends periodic heartbeats to indicate its liveness "
+            + "to the broker. If no heartbeats are received by the broker before the expiration of this session timeout, "
+            + "then the broker will remove this client from the group and initiate a rebalance. Note that the value "
+            + "must be in the allowable range as configured in the broker configuration by <code>group.min.session.timeout.ms</code> "
+            + "and <code>group.max.session.timeout.ms</code>.";
 
     public static final String HEARTBEAT_INTERVAL_MS_CONFIG = "heartbeat.interval.ms";
     public static final String HEARTBEAT_INTERVAL_MS_DOC = "The expected time between heartbeats to the consumer "
-                                                           + "coordinator when using Kafka's group management facilities. Heartbeats are used to ensure that the "
-                                                           + "consumer's session stays active and to facilitate rebalancing when new consumers join or leave the group. "
-                                                           + "The value must be set lower than <code>session.timeout.ms</code>, but typically should be set no higher "
-                                                           + "than 1/3 of that value. It can be adjusted even lower to control the expected time for normal rebalances.";
+            + "coordinator when using Kafka's group management facilities. Heartbeats are used to ensure that the "
+            + "consumer's session stays active and to facilitate rebalancing when new consumers join or leave the group. "
+            + "The value must be set lower than <code>session.timeout.ms</code>, but typically should be set no higher "
+            + "than 1/3 of that value. It can be adjusted even lower to control the expected time for normal rebalances.";
+
+    public static final String DEFAULT_LIST_KEY_SERDE_INNER_CLASS = "default.list.key.serde.inner";
+    public static final String DEFAULT_LIST_KEY_SERDE_INNER_CLASS_DOC = "Default inner class of list serde for key that implements the <code>org.apache.kafka.common.serialization.Serde</code> interface.";
+
+    public static final String DEFAULT_LIST_VALUE_SERDE_INNER_CLASS = "default.list.value.serde.inner";
+    public static final String DEFAULT_LIST_VALUE_SERDE_INNER_CLASS_DOC = "Default inner class of list serde for value that implements the <code>org.apache.kafka.common.serialization.Serde</code> interface.";
+
+    public static final String DEFAULT_LIST_KEY_SERDE_TYPE_CLASS = "default.list.key.serde.type";
+    public static final String DEFAULT_LIST_KEY_SERDE_TYPE_CLASS_DOC = "Default class for key that implements the <code>java.util.List</code> interface. "
+            + "Note when list serde class is used, one needs to set the inner serde class that implements the <code>org.apache.kafka.common.serialization.Serde</code> interface via '"
```

Comment:
`Note when list serde class is used` -> seems be a little fuzzy if one does not know the context. It might be better to be very explicit.

What about:
```
"Default class for keys (that are of type <code>java.util.List</code>) that implements the <code>java.util.List</code> interface. "
+ "This config is only effective if configuration parameter " + DEFAULT_KEY_SERDE_CLASS_CONFIG + " is set to " + ListSerde.class.getName() + " and determines the concrete List-type (eg, LinkedList, ArrayList etc.) that is instantiated during deserialization. "
+ "Note that you also need to set the inner serde class (for the list elements) that implements the <code>org.apache.kafka.common.serialization.Serde</code> interface via " + DEFAULT_LIST_KEY_SERDE_INNER_CLASS + " configuration parameter."
```

We should be similarly explicit, for the "inner serde" configs.

Reply:
`DEFAULT_KEY_SERDE_CLASS_CONFIG` is not accessible from `ComminClientConfigs` since it lives in `StreamsConfig`, the same applies for `ListSerde.class.getName()`